src/test/java/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterDefinitionTest.java

```diff
@@ -659,4 +673,38 @@ public void testSafeDescription() throws Exception {
         assertNotNull(page.getElementById("test-expected"));
         assertNull(page.getElementById("test-not-expected"));
     }
+
+    /** Upgrade from settings stored prior to 1.1.0. */
+    @Issue("JENKINS-49573")
+    @LocalData
+    @Test
+    public void compat() throws Exception {
+        MatrixProject p = j.jenkins.getItemByFullName("p", MatrixProject.class);
+        MatrixBuild b = j.assertBuildStatusSuccess(p.scheduleBuild2(0, (Cause) null, new ParametersAction(new MatrixCombinationsParameterValue("x", "", Arrays.asList("x=b,y=2")))));
+        b.save();
+        System.out.println(new XmlFile(Run.XSTREAM, new File(b.getRootDir(), "build.xml")).asString());
+        p.save();
+        System.out.println(p.getConfigFile().asString());
+    }
+
+    /** Settings stored prior to 1.1.0 but job then disabled and reënabled <em>without</em> resaving the configuration form. */
+    @Issue("JENKINS-49573")
+    @LocalData
+    @Test
+    public void hairyCompat() throws Exception {
+        for (Map.Entry<Saveable, OldDataMonitor.VersionRange> entry : AdministrativeMonitor.all().get(OldDataMonitor.class).getData().entrySet()) {
+            // as of JEP-200 will show errors from com.google.common.collect.Lists$TransformingRandomAccessList
+            System.out.println(entry.getKey() + " → " + entry.getValue().extra);
+        }
+        MatrixProject p = j.jenkins.getItemByFullName("p", MatrixProject.class);
+        MatrixBuild b = j.assertBuildStatusSuccess(p.scheduleBuild2(0, (Cause) null, new ParametersAction(new MatrixCombinationsParameterValue("x", "", Arrays.asList("x=b,y=2")))));
+        for (MatrixCombinationsShortcut shortcut : ((MatrixCombinationsParameterDefinition) p.getProperty(ParametersDefinitionProperty.class).getParameterDefinition("x")).getShortcutList()) {
+            System.out.println(shortcut.getCombinationsData(p, b));
+        }
+        b.save();
+        System.out.println(new XmlFile(Run.XSTREAM, new File(b.getRootDir(), "build.xml")).asString());
+        p.save();
+        System.out.println(p.getConfigFile().asString());
+    }

```

Comment:
The output: https://gist.github.com/ikedam/d9ec50761d4c0f2020662f33606887aa

Reply:
What happens:
* Fail to deserialize `resultsToCheck#c` for JEP-200. It resuluts broken `resultsToCheck`.
* Somehow it causes NPE in `shourcutList#a`. It results `shortcutList#a` to be `null`.
* It results `shortcutList#toString()` fail, and `readResolve` sets a default value for `shortcutList`.

Reply:
Right. Not the cleanest fix, but probably good enough.